eth-types/src/revm.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    l2_types::{BlockTrace, TransactionTrace},
+    ToBigEndian,
+};
+use revm::primitives::{Address, BlockEnv, CreateScheme, TransactTo, TxEnv, B256, U256};
+
+impl From<&BlockTrace> for BlockEnv {
+    fn from(block: &BlockTrace) -> Self {
+        BlockEnv {
+            number: U256::from(block.header.number.unwrap().as_u64()),
+            coinbase: Address::from(block.coinbase.address.unwrap().to_fixed_bytes()),
+            timestamp: U256::from_be_bytes(block.header.timestamp.to_be_bytes()),
+            gas_limit: U256::from_be_bytes(block.header.gas_limit.to_be_bytes()),
+            basefee: U256::from_be_bytes(
+                block
+                    .header
+                    .base_fee_per_gas
+                    .unwrap_or_default()
+                    .to_be_bytes(),
+            ),
+            difficulty: U256::from_be_bytes(block.header.difficulty.to_be_bytes()),
+            prevrandao: block
+                .header
+                .mix_hash
+                .map(|h| B256::from(h.to_fixed_bytes())),
+            blob_excess_gas_and_price: None,
+        }
+    }
+}
+
+impl From<&TransactionTrace> for TxEnv {
+    fn from(tx: &TransactionTrace) -> Self {
```

Comment:
i see.. so here is the bridge/boudary between old ethers-rs and the new alloy-rs.  Not very complex.

Reply:
It's not exist for alloy-rs, it's part of revm even in old version